server-spi-private/src/main/java/org/keycloak/email/EmailTemplateProvider.java

```diff
@@ -76,4 +77,24 @@
 
     public void sendVerifyEmail(String link, long expirationInMinutes) throws EmailException;
 
+    /**
+     * Send formatted email
+     *
+     * @param subjectFormatKey message property that will be used to format email subject
+     * @param bodyTemplate freemarker template file
+     * @param bodyAttributes attributes used to fill template
+     * @throws EmailException
+     */
+    void send(String subjectFormatKey, String bodyTemplate, Map<String, Object> bodyAttributes) throws EmailException;
```

Comment:
Why are changes to the EmailProvider required for the text authentication work?

Reply:
Because users have asked for a generic email sender.  And, the text based verify email uses a code rather than a link so needs a different template than the UI based one.

Reply:
As the user would read the email in the UI and not on the console I don't see why this shouldn't just send a regular link that is opened in the browser to confirm the email address. Having to copy/paste a code to the command line would be more awkward to the user.

Reply:
I don't agree. Sorry.